examples/plot_ranking.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) Fairlearn contributors.
```

Comment:
I'm a bit surprised we're adding an example, but not a user guide section. Sometimes the latter can borrow from the former through `literalinclude`, but in my mind the first step is always the user guide. That said, I don't think we're making that distinction particularly clear at the moment, which is probably something to discuss on the community call again (topic: "structure of documentation" with a particular focus on user guide vs. examples)